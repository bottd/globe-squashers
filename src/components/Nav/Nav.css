.Nav {
  background-color: #000;
  display: flex;
  position: fixed;
  top: 1px;
  width: 100vw;
  z-index: 999;
}

.NavLink {
  color: #FFF;
  font-size: 2em;
  margin-left: 1em;
  text-decoration: none;
}

.NavLink:hover {
  text-decoration: underline;
}

.active {
  text-decoration: underline;
}
