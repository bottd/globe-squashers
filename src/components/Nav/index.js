import React from 'react';
import {NavLink} from 'react-router-dom';
import './Nav.css';

export default function Nav() {
  return (
    <div className="Nav">
      <NavLink exact to="/" className="NavLink">
        HOME
      </NavLink>
      <NavLink to="/create" className="NavLink">
        CREATE
      </NavLink>
      <NavLink to="/browse" className="NavLink">
        BROWSE
      </NavLink>
    </div>
  );
}
