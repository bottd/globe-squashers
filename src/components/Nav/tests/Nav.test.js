import React from 'react';
import Nav from '../';
import {shallow} from 'enzyme';

describe('Nav', () => {
  it('Should render like snapshot', () => {
    const wrapper = shallow(<Nav/>);
    expect(wrapper).toMatchSnapshot();
  });
});
