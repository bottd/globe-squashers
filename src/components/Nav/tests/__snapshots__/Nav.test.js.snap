// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          HOME
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/create"
        >
          CREATE
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/browse"
        >
          BROWSE
        </NavLink>,
      ],
      "className": "Nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "HOME",
          "to": "/",
        },
        "ref": null,
        "rendered": "HOME",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "CREATE",
          "to": "/create",
        },
        "ref": null,
        "rendered": "CREATE",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "BROWSE",
          "to": "/browse",
        },
        "ref": null,
        "rendered": "BROWSE",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            HOME
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/create"
          >
            CREATE
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/browse"
          >
            BROWSE
          </NavLink>,
        ],
        "className": "Nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "HOME",
            "to": "/",
          },
          "ref": null,
          "rendered": "HOME",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "CREATE",
            "to": "/create",
          },
          "ref": null,
          "rendered": "CREATE",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "BROWSE",
            "to": "/browse",
          },
          "ref": null,
          "rendered": "BROWSE",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
